src/mlpack/methods/ann/modelParser.cpp

```diff
@@ -0,0 +1,1132 @@
+    * @param trainY Correct output/labels for the training input data
+    */
+   Dataset(arma::mat& trainX, arma::mat& trainY)
+   {
+     this->trainX = trainX;
+     this->trainY = trainY;
+   }
+   /**
+    * Create the Dataset object
+    * 
+    * Pass the training input dataset, validation input dataset and their
+    * respective output datasets
+    * 
+    * @param trainX Training input
+    * @param trainY Correct ouput/labels for the training input data
+    * @param validX Validation input
+    * @param validY Correct output/labels for the validation input data
+    */
+   Dataset(arma::mat& trainX, arma::mat& trainY,
+           arma::mat& validX, arma::mat& validY)
+   {
+     this->trainX = trainX;
+     this->trainY = trainY;
+     this->validX = validX;
+     this->validY = validY;
+   }
+   /**
+    * Set the values of the training dataset and its corresponding output
+    * 
+    * Pass the training input dataset and its corresponding output dataset
+    * 
+    * @param trainX Training input
+    * @param trainY Correct output/labels for the training input data
+    */
+   void setTrainSet(arma::mat& trainX, arma::mat& trainY)
+   {
+     this->trainX = trainX;
+     this->trainY = trainY;
+   }
+   /**
+    * Set the values of the validation dataset and its corresponding output
+    * 
+    * Pass the validation input dataset and its corresponding output dataset
+    * 
+    * @param validX Validation input
+    * @param validY Correct output/labels for the validation input data
+    */
+   void setValidSet(arma::mat& validX, arma::mat& validY)
+   {
+     this->validX = validX;
+     this->validY = validY;
+   }
+   /**
+    * Return the values of the training input dataset
+    */
+   arma::mat getTrainX()
+   {
+     return trainX;
+   }
+   /**
+    * Return the values of the training output dataset
+    */
+   arma::mat getTrainY()
+   {
+     return trainY;
+   }
+   /**
+    * Return the values of the validation input dataset
+    */
+   arma::mat getValidX()
+   {
+     return validX;
+   }
+   /**
+    * Return the values of the validation output dataset
+    */
+   arma::mat getValidY()
+   {
+     return validY;
+   }
+};
+
+/**
+ * Print the given stl map where the keys are of type string and
+ * values are of type double
+ * 
+ * @param params The map to be printed
+ */
+void printMap(map<string, double> params)
+{
+  map<string, double>::iterator itr;
+  for (itr = params.begin(); itr != params.end(); ++itr)
+  {
+    cout << itr->first << " : " << itr->second << endl;
+  }
+}
+
+/**
+ * Update the values of a given stl map with that of another map 
+ * corresponding to the keys that are common
+ * 
+ * Keys are of type string and values are of type double
+ * 
+ * @param origParams The map whose values will be updated
+ * @param newParams The map whose values will be used to update origParams
+ */
+void updateParams(map<string, double> &origParams, map<string, double> &newParams)
```

Comment:
You can use a copy constructor instead or directly assign values to optimizerparam instead of copying in loop.

Reply:
Thanks for the suggestion. That might be a better solution, though I will have to look into it a bit more. One reason why it does what it does is that I wanted to check for errors. For example, some keys are absolutely necessary, some are not. That validation part is not done yet. I will keep this in mind while implementing the validation process.